woocommerce/woocommerce
# Add/28568 dashboard widget - setup

### All Submissions:






### Changes proposed in this Pull Request:



Closes #28568 

This PR adds "WooCommerce Setup" widget to promote users to finish the onboarding tasks. The widget shows only when there are unfinished tasks.

This PR also hides the existing "WooCommerce Status" widget until the store owner completes the onboarding tasks.

### How to test the changes in this Pull Request:

1. Start with a fresh installation of WooCoommerce and don't start onboarding wizard yet.
2. Navigate to WordPress Dashboard and confirm that you see a new widget with the title "WooCommerce Setup", but "WooCommece Status" widget is hidden. 
3. Confirm that the widget shows "Step 0 of 5" (the total # of steps can be 6 if you have a payment plugin available)
4. Start working on onboarding tasks one by one.
5. As you finish onboarding tasks, check back and confirm that step # increases. 
6. The "WooCommerce Setup" widget should not be displayed once you complete the onboarding tasks, but "WooCommerce Status" widget should be displayed again.


### Screenshots
![Screen Shot 2021-02-18 at 2 50 49 PM](https://user-images.githubusercontent.com/4723145/108432105-b6f0e200-71f8-11eb-8b02-6465e00bf854.jpg)


## Note

I've noticed that we have the onboarding tasks defined in multiple places now. It might not be a big deal, but we have to update two places when we add a new onboarding task. It would be nice if we have an API where we can get a list of onboarding tasks. I will try to see if it's worth working on that during the upcoming cooldown period. 

[tasks defined in js
](https://github.com/woocommerce/woocommerce-admin/blob/main/client/task-list/tasks.js)

[tasks defined in this PR](https://github.com/woocommerce/woocommerce/pull/29174/files#diff-857cada53ba223723d6caaabeaddb6909539ac1a748781e4718d6f36b9c03172R27)

### Changelog entry

> Enhancement - Add a new dashboard widget to promote users to finish onboarding tasks.
